File: scratch/parse_bing.py
import re

with open('scratch/bing_test_loose.html', 'r', encoding='utf-8') as f:
    html = f.read()

# Regular expression to find <li class="b_algo"> ... </li>
b_algo_blocks = re.findall(r'<li[^>]*class="b_algo"[^>]*>(.*?)</li>', html, re.DOTALL)

print(f"Found {len(b_algo_blocks)} b_algo blocks.")

for i, block in enumerate(b_algo_blocks, 1):
    print(f"\n--- Result #{i} ---")
    
    # Try to find the link and title
    # Links usually look like <a href="URL" ...>Title</a> or similar
    # Inside h2 or direct
    link_match = re.search(r'<a[^>]*href="([^"]+)"[^>]*>(.*?)</a>', block, re.DOTALL)
    if link_match:
        url = link_match.group(1)
        title_raw = link_match.group(2)
        title = re.sub(r'<[^>]+>', '', title_raw).strip()
        print(f"URL: {url}")
        print(f"Title: {title}")
    else:
        print("No link matched")
        
    # Try to find the snippet
    snippet_match = re.search(r'<p[^>]*>(.*?)</p>', block, re.DOTALL)
    if snippet_match:
        snippet_raw = snippet_match.group(1)
        snippet = re.sub(r'<[^>]+>', '', snippet_raw).strip()
        print(f"Snippet: {snippet}")

File: scratch/parse_bing.js
const fs = require('fs');

const html = fs.readFileSync('scratch/bing_test_loose.html', 'utf8');

// Extrai blocos <li class="b_algo">...</li>
const bAlgoRegex = /<li[^>]*class="b_algo"[^>]*>([\s\S]*?)<\/li>/g;
let match;
let count = 0;

console.log('--- Parse Bing Results ---');
while ((match = bAlgoRegex.exec(html)) !== null) {
  count++;
  const block = match[1];
  
  // Extrai o link principal
  // Geralmente: <a ... href="URL" ...>TITLE</a>
  const linkRegex = /<a[^>]*href="([^"]+)"[^>]*>([\s\S]*?)<\/h2>/i;
  const linkMatch = linkRegex.exec(block);
  
  if (linkMatch) {
    const rawUrl = linkMatch[1];
    let title = linkMatch[2].replace(/<[^>]+>/g, '').trim();
    
    // Decodifica a URL se for uma URL de redirecionamento do Bing
    let url = rawUrl;
    if (url.includes('bing.com/ck/a')) {
      // Tenta extrair a URL real do atributo u ou de outros parâmetros se visível,
      // ou apenas usa como está. Mas geralmente o href real também está no atributo "u" ou "h"
      // Vamos tentar encontrar a URL limpa se houver.
    }
    
    // Extrai o snippet
    // Geralmente em um <p class="b_algo_text" ou similar
    const snippetRegex = /<p[^>]*>([\s\S]*?)<\/p>/i;
    const snippetMatch = snippetRegex.exec(block);
    const snippet = snippetMatch ? snippetMatch[1].replace(/<[^>]+>/g, '').trim() : '';
    
    console.log(`\n[Result #${count}]`);
    console.log(`URL: ${url}`);
    console.log(`Title: ${title}`);
    console.log(`Snippet: ${snippet}`);
  } else {
    // Tenta outro regex simples
    const simpleLink = /href="([^"]+)"[^>]*>([\s\S]*?)<\/a>/i.exec(block);
    if (simpleLink) {
      console.log(`\n[Result #${count} (simple)]`);
      console.log(`URL: ${simpleLink[1]}`);
      console.log(`Title: ${simpleLink[2].replace(/<[^>]+>/g, '').trim()}`);
    } else {
      console.log(`\n[Result #${count} (no link matched)]`);
    }
  }
}
console.log(`\nTotal parsed: ${count}`);
